torch/_prims_common/wrappers.py

```diff
@@ -157,9 +157,17 @@ def _resize_output_check(out: TensorLikeType, shape: ShapeType):
 
 
 # TODO: handle tuples of tensors
-def _maybe_resize_out(out: TensorLikeType, shape: ShapeType):
+def _maybe_resize_out(
+    out: TensorLikeType,
+    shape: ShapeType,
+    stride: Optional[Union[List[int], int]] = None,
+    is_copy: bool = False
+) -> TensorLikeType:
     if _resize_output_check(out, shape):
-        return out.resize_(shape)
+        out = out.resize_(shape)
+        if stride is not None and not is_copy:
+            out = out.as_strided_(shape, stride)
+        return out
```

Comment:
Updated:
- checks if `out` can be safely resized (via `_resize_output_check `) and only re-strides in that case
- skips `*_copy` functions (e.g., `diagonal_copy`).

When trying to use `as_strided_` with `diagonal_copy`, I ran into an issue with not enough storage being available for a certain combination of shape + stride + dtype. The following seems to work, but it's not in-place, so seems problematic, so I ended up just skipping copy funcs.

```py
out = torch.empty_strided(shape, stride, dtype=out.dtype, device=out.device)
```

See: https://github.com/pytorch/pytorch/wiki/Developer-FAQ#how-does-out-work-in-pytorch

Reply:
The fact that these are failing shows a set of functions for which this logic is just not correct. Please do not add skips for the sake of it, as it may mask errors, as in this case.